skimage/feature/_basic_features.py

```diff
@@ -0,0 +1,172 @@
+    img = np.ascontiguousarray(img_as_float32(img))
+    if num_sigma is None:
+        num_sigma = int(np.log2(sigma_max) - np.log2(sigma_min) + 1)
+    sigmas = np.logspace(
+        np.log2(sigma_min),
+        np.log2(sigma_max),
+        num=int(np.log2(sigma_max) - np.log2(sigma_min) + 1),
+        base=2,
+        endpoint=True,
+    )[::-1]
+    with ThreadPoolExecutor(max_workers=num_workers) as ex:
+        out_sigmas = list(
+            ex.map(
+                lambda s: _singlescale_basic_features_singlechannel(
+                    img, s, intensity=intensity, edges=edges, texture=texture
+                ),
+                sigmas,
+            )
+        )
+    features = itertools.chain.from_iterable(out_sigmas)
+    return features
+
+
+def multiscale_basic_features(
+    image,
+    multichannel=False,
+    intensity=True,
+    edges=True,
+    texture=True,
+    sigma_min=0.5,
+    sigma_max=16,
+    num_sigma=None,
+    num_workers=None,
+):
+    """Local features for a single- or multi-channel nd image.
+
+    Intensity, gradient intensity and local structure are computed at
+    different scales thanks to Gaussian blurring.
+
+    Parameters
+    ----------
+    image : ndarray
+        Input image, which can be grayscale or multichannel.
+    multichannel : bool, default False
+        True if the last dimension corresponds to color channels.
+    intensity : bool, default True
+        If True, pixel intensities averaged over the different scales
+        are added to the feature set.
+    edges : bool, default True
+        If True, intensities of local gradients averaged over the different
+        scales are added to the feature set.
+    texture : bool, default True
+        If True, eigenvalues of the Hessian matrix after Gaussian blurring
+        at different scales are added to the feature set.
+    sigma_min : float, optional
+        Smallest value of the Gaussian kernel used to average local
+        neighbourhoods before extracting features.
+    sigma_max : float, optional
+        Largest value of the Gaussian kernel used to average local
+        neighbourhoods before extracting features.
+    num_sigma : int, optional
+        Number of values of the Gaussian kernel between sigma_min and sigma_max.
+        If None, sigma_min multiplied by powers of 2 are used.
+    num_workers : int or None, optional
+        The number of parallel threads to use. If set to ``None``, the full
+        set of available cores are used.
+
+
+    Returns
+    -------
+    features : np.ndarray
+        Array of shape ``image.shape + (n_features,)``
+    """
+    if not any([intensity, edges, texture]):
+        raise ValueError(
+                "At least one of ``intensity``, ``edges`` or ``textures``"
+                "must be True for features to be computed."
+                )
+    if image.ndim < 3:
+        multichannel = False
+    if not multichannel:
+        image = image[..., np.newaxis]
```

Comment:
```suggestion
    image = np.atleast_3d(image)
```

Reply:
hum, I think this would not work for a 3D image (with 3 spatial dimensions)

Reply:
yes, I see!